crates/ide_db/src/defs.rs

```diff
@@ -302,17 +312,25 @@ impl NameClass {
 #[derive(Debug)]
 pub enum NameRefClass {
     Definition(Definition),
-    FieldShorthand { local_ref: Local, field_ref: Definition },
+    FieldShorthand { local_ref: Local, field_ref: Field },
 }
 
 impl NameRefClass {
-    /// `Definition`, which this name refers to.
-    pub fn referenced(self) -> Definition {
+    /// `Definition`, which this name refers to with a preference for the field reference in case of a field shorthand.
+    pub fn referenced_field(self) -> Definition {
```

Comment:
Hm, not sure if that's the best API: `or_field` feels too specific. I wonder if the best solution here is maybe not to have any methods at all, and just write explicit matches at every call-site? That's more code, but it makes it clear when we ignore some cases...

Reply:
Ye the main reason I didn't feel like doing that was due to the extern crate variant which you conveniently got rid of now. So inling all of these might be fine now.